astropy/time/formats.py

```diff
@@ -547,7 +547,7 @@ def _check_val_type(self, val1, val2):
             val1.dtype if orig_val2_is_none else np.result_type(val1.dtype, val2.dtype)
         )
         subfmts = self._select_subfmts(self.in_subfmt)
-        for subfmt, dtype, convert, _ in subfmts:
+        for _, dtype, convert, _ in subfmts:  # noqa: B007
```

Comment:
post mortem, but for later reference: you can avoid a noqa here *and* keep a clear, describing variable name as long as you prepend a `_` to it
```suggestion
        for _subfmt, dtype, _convert, _ in subfmts:
```

Reply:
oh, I missed that ! in that case a noqa seems like the lesser of two evils